recipe-ts-runtime/src/Ingredient.ts

```diff
@@ -63,6 +63,12 @@ export abstract class Ingredient {
         }
     }
 
+    protected duplicate(): Ingredient {
+        const copy = Object.assign(Object.create(Object.getPrototypeOf(this)), this);
+        copy.properties = new Map(this.properties);
```

Comment:
Are any of the properties stateful and need copying?

Reply:
Nope, just want a new copy of the map.